src/query/procedure/mg_procedure_impl.cpp

```diff
@@ -3322,6 +3335,107 @@ mgp_error mgp_graph_delete_edge(struct mgp_graph *graph, mgp_edge *edge) {
   });
 }
 
+mgp_error mgp_graph_has_text_index(mgp_graph *graph, const char *index_name, int *result) {
+  return WrapExceptions([graph, index_name, result]() {
+    std::visit(memgraph::utils::Overloaded{
+                   [&](memgraph::query::DbAccessor *impl) { *result = impl->TextIndexExists(index_name); },
+                   [&](memgraph::query::SubgraphDbAccessor *impl) {
+                     *result = impl->GetAccessor()->TextIndexExists(index_name);
+                   }},
+               graph->impl);
+  });
+}
+
+mgp_vertex *GetVertexByGid(mgp_graph *graph, memgraph::storage::Gid id, mgp_memory *memory) {
+  std::optional<memgraph::query::VertexAccessor> maybe_vertex =
+      std::visit([graph, id](auto *impl) { return impl->FindVertex(id, graph->view); }, graph->impl);
+  if (maybe_vertex) {
+    return std::visit(memgraph::utils::Overloaded{
+                          [memory, graph, maybe_vertex](memgraph::query::DbAccessor *) {
+                            return NewRawMgpObject<mgp_vertex>(memory, *maybe_vertex, graph);
+                          },
+                          [memory, graph, maybe_vertex](memgraph::query::SubgraphDbAccessor *impl) {
+                            return NewRawMgpObject<mgp_vertex>(
+                                memory, memgraph::query::SubgraphVertexAccessor(*maybe_vertex, impl->getGraph()),
+                                graph);
+                          }},
+                      graph->impl);
+  }
+  return nullptr;
+}
+
+void WrapTextSearch(std::vector<memgraph::storage::Gid> vertex_ids, std::string error_msg, mgp_graph *graph,
```

Comment:
One of the user stories was to return granular error messages (see [Notion](https://www.notion.so/memgraph/Make-sure-to-return-more-granular-error-message-from-the-text-search-index-tantivy-to-the-user-6f2a6fce21864bbe8d50b8fc98cd2d65)), and that would necessarily entail returning a large variety of error messages (e.g. for syntactically incorrect Tantivy queries).
Since you can’t pass exceptions through the underlying C API, the alternative was to return the error message in the result.

Reply:
but could you make use of wrap exceptions and mgp::error?

Reply:
I really don't have any. Since you need to return result and error message, I don't see how to do it without hacking it with mgp_map